packages/react/src/analytics/integrations/Castle/CastleV2.js

```diff
@@ -0,0 +1,328 @@
+   *
+   * @async
+   *
+   * @param {object} config - Axios config object.
+   *
+   * @returns {object} - The modified Axios config object.
+   */
+  onBeforeRequestFullfil = async config => {
+    const requestTokenValue = await this.castleJS.createRequestToken();
+
+    config.headers[this.clientIdHeaderName] = requestTokenValue;
+
+    return config;
+  };
+
+  /**
+   * Callback that is used on the Axios interceptor to handle any error when setting the Castle header.
+   *
+   * @async
+   *
+   * @param {object|string} error - The error itself.
+   *
+   * @returns {Promise} - The rejected promise.
+   */
+  onBeforeRequestReject = async error => {
+    return Promise.reject(error);
+  };
+
+  /**
+   * @param {object} data - Data Track event data.
+   *
+   * @returns {object} - The formatted user object.
+   */
+  getUserData(data) {
+    const userData = data.user || {};
+    const userTraits = data.user.traits || {};
+
+    const formattedUserData = {
+      id: userData.id,
+      email: userTraits.email,
+      phone: userTraits.phoneNumber,
+      name: userTraits.name,
+      registered_at: userData.traits.createdDate,
+      traits: {
+        isGuest: userData.traits.isGuest,
+      },
+    };
+
+    // clean "falsy" values to ensure the SDK accepts the request.
+    return pickBy(formattedUserData, identity);
+  }
+
+  /**
+   * Track method that will filter by event type: page or track.
+   *
+   * @param {object} data Track event data.
+   *
+   * @async
+   *
+   * @returns {Promise} Promise that will resolve when the method finishes.
+   */
+  async track(data) {
+    switch (data.type) {
+      case trackTypes.PAGE:
+        return this.trackPage(data);
+
+      case trackTypes.TRACK:
+        return this.trackEvent(data);
+      /* istanbul ignore next */
+      default:
+        return;
+    }
+  }
+
+  /**
+   * Method that will handle each type of event to be tracked.
+   *
+   * @param {object} data - Data Track event data.
+   *
+   * @returns {Promise<Castle>} - The resolved promise of each castle call method.
+   */
+  async trackEvent(data) {
+    const user = this.getUserData(data);
+
+    switch (data.event) {
+      case eventTypes.ADDRESS_INFO_ADDED:
+      case eventTypes.CHECKOUT_STEP_COMPLETED:
+      case eventTypes.LOGIN:
+      case eventTypes.PAYMENT_INFO_ADDED:
+      case eventTypes.PROMOCODE_APPLIED:
+      case eventTypes.SHIPPING_INFO_ADDED:
+      case eventTypes.SHIPPING_METHOD_ADDED:
+      case eventTypes.SIGNUP_FORM_COMPLETED:
+      case eventTypes.SIGNUP_NEWSLETTER: {
+        const formData = {
+          user,
+          name: data.event,
+          values: data.properties,
+        };
+
+        return this.castleJS.form(formData).then(
+          () =>
+            this.debugModeOn &&
+            utils.logger.info(
+              `${CASTLE_MESSAGE_PREFIX} Form track success. Payload:`,
+              formData,
+            ),
+          error => Promise.reject(error),
```

Comment:
This is the same case, you can remove this function argument and the promise will be rejected automatically.

```js
return this.castleJS.form(formData).then(
          () =>
            this.debugModeOn &&
            utils.logger.info(
              `${CASTLE_MESSAGE_PREFIX} Form track success. Payload:`,
              formData,
            ),
         );
    }
```